Sources/VLCSettingsController.m

```diff
@@ -192,9 +181,7 @@ - (void)updateForPasscode:(NSString *)passcode
     [VLCKeychainCoordinator setPasscodeWithPasscode:passcode error:&error];
     if (error == nil) {
         //Set manually the value to enable/disable the UISwitch.
-        BOOL passcodeEnabled = passcode != nil;
-        [[NSUserDefaults standardUserDefaults] setBool:passcodeEnabled forKey:kVLCSettingPasscodeOnKey];
-        [self updateUIAndCoreSpotlightForPasscodeSetting:passcode != nil];
```

Comment:
turns out the default gets already set by the settingskit (noticed that because the observer was called twice) and the deletion is now handled in medialibrary so this method can go :)